Update: Introduce usecase for customer and reminder. Fixed json mulfo…

Dev-Pasaka opened the pull request:
…rmation in test cases

Commit:
Update: Introduce usecase for customer and reminder. Fixed json mulformation in test cases

Dev-Pasaka commented:
i have introduced use case in customers and reminders. i also fixed some json decoding issues and json mulformation  in ApplicationTests

antonarhipov commented on src/main/kotlin/domain/customer/usecases/AddContactUseCase.kt:
```diff
@@ -0,0 +1,27 @@
+    private val customerRepository: CustomerRepository,
+    private val eventPublisher: EventPublisher
+) {
+    operator fun invoke(customerId: CustomerId, contact: Contact):Customer?{
```
@Dev-Pasaka What is the benefit of overriding the operator here if we call the function explicitly anyway? I'd rather make this a normal function so the example wouldn't promote operator overloading without necessity.

Commit:
Refactor: Replace operator  with explicit  method in use cases

- Replaced all  declarations with
- Updated service classes to call  instead of
- Aligns with review recommendation to avoid unnecessary operator overloading
- Improves readability and avoids promoting patterns without clear necessity
